### PR TITLE
[Pipeline] Require specified build from 'azure-powershell - sign' to download

### PR DESCRIPTION
**Pull request by BethanyZhou**
## Description

Before: Always consume the newest build artifacts of 'azure-powershell - sign'

Now: Get the build artifacts by pipeline id.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by isra-fel** [comment]
@BethanyZhou  is this PR ready for review and shall I review it?

**Comment by BethanyZhou** [comment]
Yes, please help to review it.

File: .azure-pipelines/release-test.yml [hunk 1]
```diff
@@ -1,178 +1,174 @@
...
-    filePath: 'tools/Test/SmokeTest/PrepareRequiredPowershell.ps1'
-    arguments: '-RequiredPsVersion $(PSVersion)'      
+  - task: DownloadPipelineArtifact@2
+    condition: and(succeeded(), eq(variables['GalleryName'], 'LocalRepo'))
+    displayName: Download Artifacts from Build Pipeline
+    inputs:
+      buildType: 'specific'
+      project: $(ProjectToDownloadArtifacts)
+      definition: $(BuildPipelineName)
+      buildVersionToDownload: 'specific'
```
Comment (isra-fel) [inline .azure-pipelines/release-test.yml:47]: Is it possible to download latest build?
or is it necessary/useful/helpful to download latest build?

**Review by isra-fel — COMMENTED** [review]
Could you trigger a test run?

File: .azure-pipelines/release-test.yml (hunk 1 above)
Comment (BethanyZhou) [reply .azure-pipelines/release-test.yml:47]: 1. Sure, here is a test run: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=581795&view=results
2. It is possible to download any build you want to. If anyone want to download latest build, fill its pipeline instance ID into the variable `PipelineId` before run.

**Comment by BethanyZhou** [comment]
A running example about getting the latest artifacts is: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=593944&view=results
Get the specified artifacts:  https://dev.azure.com/azure-sdk/internal/_build/results?buildId=599860&view=results
Please take a look, thanks.

**Comment by BethanyZhou** [comment]
Retired subscription example: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=601857&view=results

File: tools/Test/SmokeTest/RmCoreSmokeTests.ps1 [hunk 2]
```diff
@@ -5,6 +5,11 @@
 # $debugState = $false
 [cmdletbinding()]
 param(
+    [string]
+    [Parameter(Mandatory = $false, Position = 0)]
```
Comment (isra-fel) [inline tools/Test/SmokeTest/RmCoreSmokeTests.ps1:9]: I think subscription ID should not be a parameter of smoke test, because

1. Smoke test is based on an asumption that you have already set up the environment.
1. `-subscriptionId` does not always work. It only works when you didn't login, which may cause confusion.

Is there any specific reason why you add it? If not, I think we can simply undo changes to this file

File: tools/Test/SmokeTest/ConnectAzAccount.ps1 [hunk 3]
```diff
@@ -10,12 +10,14 @@ param(
...
+    [Parameter(Mandatory = $true, Position = 3)]
+    $subscriptionId
 )
 
 Write-Host "Connecting Az.Account..."
 $secret = ConvertTo-SecureString -String $pwd -AsPlainText -Force
 $credential = New-Object -TypeName System.Management.Automation.PSCredential($servicePrincipal, $secret)
-Connect-AzAccount -ServicePrincipal -Credential $credential -Tenant $tenantId
-Set-AzContext -Subscription "Azure SDK Powershell Test"
-Enable-AzureRmAlias
```
Comment (isra-fel) [inline tools/Test/SmokeTest/ConnectAzAccount.ps1:21]: Why did you remove this line?

**Review by isra-fel — CHANGES_REQUESTED** [review]
Please see my inline comments. Thanks

File: tools/Test/SmokeTest/ConnectAzAccount.ps1 (hunk 3 above)
Comment (isra-fel) [reply tools/Test/SmokeTest/ConnectAzAccount.ps1:21]: Because it is always enabled in smoke test script

File: tools/Test/SmokeTest/RmCoreSmokeTests.ps1 (hunk 2 above)
Comment (isra-fel) [reply tools/Test/SmokeTest/RmCoreSmokeTests.ps1]: Subscription Id is used only when you didn't sign in before running smoke tests. That is not true. So we can safely remove this parameter

**Review by isra-fel — APPROVED** [review]
LGTM. Thanks
